pkg/virt-operator/application.go

```diff
@@ -439,14 +439,19 @@ func (app *VirtOperatorApp) Run() {
 			RetryPeriod:   app.LeaderElection.RetryPeriod.Duration,
 			Callbacks: leaderelection.LeaderCallbacks{
 				OnStartedLeading: func(ctx context.Context) {
-					metrics.SetLeader(true)
+					if err := metrics.RegisterLeaderMetrics(); err != nil {
+						golog.Fatal(err)
+					}
+					metrics.SetLeader()
 					log.Log.Infof("Started leading")
 
 					// run app
 					go app.kubeVirtController.Run(controllerThreads, stop)
 				},
 				OnStoppedLeading: func() {
-					metrics.SetLeader(false)
+					if err := metrics.UnregisterLeaderMetrics(); err != nil {
```

Comment:
@machadovilaca Actually, I'm not sure we need that. 
IIUC the leader pod stops leading when it crashes and gets deleted, and then its metrics are gone anyway. I'm not aware of cases in which the leading pod is replaced without crashing, which requires removing its metrics explicitly.

Reply:
@enp0s3, @xpivarc can you please advise here?

Reply:
The `.Fatal` 2 lines below will crash the operator.

Reply:
@machadovilaca so I think we don't need to unregister, we only needed to move where we register these metrics, wdyt?

Reply:
removed